### PR TITLE
`tensordict.stack` can propagate names

### LINKED ISSUE #1347: [FeatureRequest] `tensordict.stack` preserves `names`
## Motivation

Currently `tensordict.stack` will not propagate the `names` of the tensordicts, see below:

```python
td = TensorDict({"a": torch.ones((2, 3, 4))}, batch_size=(2,3), names=['first', 'second'])
print(tensordict.stack([td, td], dim=0).names)  # ['None', 'None', 'None'] instead of ['None', 'first', 'second']
```

## Solution

Since `tensordict.cat` does propagate names, I'm proposing to copy the same logic.

## Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)

### PR DESCRIPTION
**Pull request by louisfaury**
## Description

An attempt to solve #1347

## Motivation and Context

close #1347 

- [x] I have raised an issue to propose this change ([required](https://github.com/pytorch/tensordict/issues) for new features and bug fixes)

## Types of changes

What types of changes does your code introduce? Remove all that do not apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (update in the documentation)
- [ ] Example (update in the folder of examples)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/pytorch/tensordict/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly (*required for a bug fix or a new feature*).
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by vmoens** [comment]
Took care of it in #1348

File: tensordict/_torch_func.py [hunk 1]
```diff
@@ -675,20 +675,32 @@ def stack_fn(key, values, is_not_init, is_tensor):
...
             if is_tc and not is_tensorclass(result):
                 return clz._from_tensordict(result)
             return result
         else:
+
+            if (names := list_of_tensordicts[0]._has_names()) is not None:
+                names = list_of_tensordicts[0].names
+                names.insert(dim, None)
+
             out = LazyStackedTensorDict(
```
Comment (vmoens) [inline tensordict/_torch_func.py:700]: I think here lazy stack will preserve the names, you can add one name for the new dim though
